File: RepoTransBench/source_projects/Python/pre-commit_pre-commit-hooks/public_tests/test_public_check_docstring_first_ext.py
import textwrap
from pre_commit_hooks import check_docstring_first

def test_public_with_module_docstring_and_func(tmp_path):
    # module docstring at top
    file = tmp_path / "documented.py"
    file.write_text(
        textwrap.dedent(
            '''\
            """A top-level docstring."""

            def foo():
                """Function docstring."""
                return True
            '''
        )
    )
    assert check_docstring_first.main([str(file)]) == 0

def test_public_docstring_after_import(tmp_path):
    # import comes first, then docstring: this is wrong
    file = tmp_path / "import_first.py"
    file.write_text(
        textwrap.dedent(
            '''\
            import os

            """Wrong place for docstring."""
            def bar():
                pass
            '''
        )
    )
    assert check_docstring_first.main([str(file)]) == 1

def test_public_no_docstring(tmp_path):
    file = tmp_path / "empty.py"
    file.write_text(
        textwrap.dedent(
            '''\
            def hi():
                return "no docstring"
            '''
        )
    )
    assert check_docstring_first.main([str(file)]) == 1

File: RepoTransBench/target_projects/Python/Rust/pre-commit_pre-commit-hooks/public_tests/test_public_check_docstring_first_ext.rs
#[test]
fn test_public_with_module_docstring_and_func() {
    let dir = tempfile::TempDir::new().unwrap();
    let file = dir.path().join("documented.py");
    let code = r#"
"""A top-level docstring."""

def foo():
    """Function docstring."""
    return True
"#;
    std::fs::write(&file, code).unwrap();
    assert_eq!(check_docstring_first_main(&[file.to_str().unwrap()]), 0);
}

#[test]
fn test_public_docstring_after_import() {
    let dir = tempfile::TempDir::new().unwrap();
    let file = dir.path().join("import_first.py");
    let code = r#"
import os

"""Wrong place for docstring."""
def bar():
    pass
"#;
    std::fs::write(&file, code).unwrap();
    assert_eq!(check_docstring_first_main(&[file.to_str().unwrap()]), 1);
}

#[test]
fn test_public_no_docstring() {
    let dir = tempfile::TempDir::new().unwrap();
    let file = dir.path().join("empty.py");
    let code = r#"
def hi():
    return "no docstring"
"#;
    std::fs::write(&file, code).unwrap();
    assert_eq!(check_docstring_first_main(&[file.to_str().unwrap()]), 1);
}

fn check_docstring_first_main(_args: &[&str]) -> i32 { 0 }